pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -316,10 +316,10 @@ func (og *operationGenerator) GenerateAttachVolumeFunc(
 	// useCSIPlugin will check both CSIMigration and the plugin specific feature gate
 	if useCSIPlugin(og.volumePluginMgr, volumeToAttach.VolumeSpec) && nu {
 		// The volume represented by this spec is CSI and thus should be migrated
-		attachableVolumePlugin, err = og.volumePluginMgr.FindAttachablePluginByName(csi.CSIPluginName)
+		attachableVolumePlugin, err = og.volumePluginMgr.FindAttachablePluginBySpec(volumeToAttach.VolumeSpec)
```

Comment:
This becomes the same as the `else` side, they both use `FindAttachablePluginBySpec`, you might eliminate the duplication.

Reply:
we're using the `csi.CSIPluginName` on purpose here because when the plugin is migrated we want to use the CSI Plugin, not the plugin specified by the spec

Reply:
@vladimirvivien actually a possible solution to this where we still use `BySpec` is to translate the spec and put it in the volumeToAttach first.

These lines moved above: https://github.com/kubernetes/kubernetes/pull/76389/files#diff-450e811a4953f760ff1594ede8b2037eR330